modules/internal/ChapelIO.chpl

```diff
@@ -198,25 +493,54 @@ module ChapelIO {
       }
     }
   }
-  
+
+
+  /*
+     An abstract base class type for classes that know how
+     to read values.
+   */
   class Reader {
+    /* returns `false` */
     proc writing param return false;
-    // if it's binary, we don't decorate class/record fields and values
+    /* returns `true` if this Reader is configured for binary I/O. */
     proc binary():bool { return false; }
-    // get other style elements
+    /* return other style elements.
+       subclasses should override this method */
+    pragma "no doc"
     proc styleElement(element:int):int { return 0; }
 
+    /* 
+       Return any saved error code.
+       Subclasses should override this method.
+     */
     proc error():syserr { return ENOERR; }
+    /*
+       Save an error code.
+       Subclasses should override this method.
+     */
     proc setError(e:syserr) { }
+    /* 
+       Clear any saved error code.
+       Subclasses should override this method.
+     */
     proc clearError() { }
   
+    /* Read a primitive type.
+       Subclasses should override this method.
+     */
     proc readPrimitive(ref x:?t) where _isIoPrimitiveTypeOrNewline(t) {
       //compilerError("Generic Reader.readPrimitive called");
       halt("Generic Reader.readPrimitive called");
     }
+    
+    /*
+       Read a sequence of bytes.
+       Subclasses should override this method.
+     */
```

Comment:
x must have _ddata type. Look for the comment "// A specialization is needed for _ddata as the value is the pointer its memory" in IO.chpl. But that needs to be cleaned up (just not during doc freeze).